presto-main/src/main/java/com/facebook/presto/util/GraphvizPrinter.java

```diff
@@ -491,7 +492,9 @@ public Void visitJoin(JoinNode node, Void context)
             }
 
             String criteria = Joiner.on(" AND ").join(joinExpressions);
-            printNode(node, node.getType().getJoinLabel(), criteria, NODE_COLORS.get(NodeType.JOIN));
+            String distributionType = node.getDistributionType().isPresent() ? node.getDistributionType().get().toString() : "UNKNOWN";
```

Comment:
Is it possible to get a unknown distribution type at this point?

Reply:
It depends on how the `GraphizPrinter` is being invoked. If it's through an EXPLAIN, then the optimizer would've reordered/rewritten any Join nodes to include the `DistributionType`.